CDC-Data-Reconciliation-Backend/server.py

```diff
@@ -216,6 +307,17 @@ def fetch_reports_from_db(report_id: int):
         print(f"Database error: {e}")
         return None
 
+def insert_report(createdDate, creationTime, noOfDiscrepancies):
+    cur.execute("INSERT INTO Reports (CreatedAtDate, TimeOfCreation, NumberOfDiscrepancies) VALUES (?, ?, ?)", (createdDate, creationTime, noOfDiscrepancies))
+    report_id = cur.lastrowid
+    app.liteConn.commit()
+    return report_id
+
+def insert_statistics(reportId, eventCode, totalCases, totalDuplicates, totalMissingFromCDC, totalMissingFromState, totalWrongAttributes):
+    cur.execute("INSERT INTO Statistics (ReportID, EventCode, TotalCases, TotalMissingFromCDC, TotalMissingFromState, TotalWrongAttributes) VALUES (?, ?, ?, ?, ?, ?)", (reportId, eventCode, totalCases, totalDuplicates, totalMissingFromCDC, totalMissingFromState, totalWrongAttributes))
```

Comment:
You should create an array of tuples to pass into the insert statement so that you are not calling insert many many times and instead you call it once. Doing it this way of doing one insert per row can have serious performance effects.

Reply:
agreed, much better to make one insertion with multiple rows than do tons of small inserts. Alongside performance it can also cause problems with being able to rollback if there are issues with so many inserts i believe.